### PR TITLE
Outstanding work

### PR DESCRIPTION
**Pull request by Iamdomchap90**
# Forecast Ticket Link(s)

https://app.forecast.it/project/P-155/task-board/T26719
https://app.forecast.it/project/P-155/task-board/T26512
https://app.forecast.it/project/P-155/task-board/T26511

### CODE REVIEW

File: checklists/views.py [hunk 1]
```diff
@@ -183,6 +190,24 @@ def edit_notes(request, checklist_id):
...
+@require_POST
+def edit_category(request, checklist_id):
+    """
+    Updates the category field on a checklist.
+    """
+
+    checklist = get_object_or_404(models.Checklist, id=checklist_id)
+
+    checklist.category = None
+    if request.POST["category"]:
```
Comment (Will-Hoey) [inline checklists/views.py:204]: If you're checking something exists in a dictionary like this you're better off using `POST.get(key)` so it doesn't fall over if it doesn't exist

File: templates/views.py [hunk 2]
```diff
@@ -119,6 +130,25 @@ def description(request, template_id):
...
+@require_POST
+def type(request, template_id):
+    """
+    This changes a template type.
+    """
+
+    template = get_object_or_404(models.Template, id=template_id)
+
+    template.type = None
+    if request.POST["type"]:
```
Comment (Will-Hoey) [inline templates/views.py:144]: As above

File: core/utils.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    message = None
+    if request.method == "POST" and "name" in request.POST:
+        form = form_class(request.POST)
+        if form.is_valid():
+            form.save()
+            message = f"'{form.cleaned_data['name']}' successfully created"
+            # clear input data upon saving category
+            form = form_class()
+
+    return [form, message]
```
Comment (Will-Hoey) [inline core/utils.py:17]: Could you outline a bit of your thinking here? Is there a reason we aren't giving this the form_class directly and retrieving it from the admin?

Comment (Iamdomchap90) [reply core/utils.py:17]: This code is resuable across Template and Checklist views hence why it is in utils. Takes the respective category admin form and compartmentalizes view logic to do with submitting this form. Abstracting the form from admin class is most DRY solution. I will leave a comment on the method.

**Review by karlswatman — APPROVED** [review]

File: core/utils.py (hunk 3 above)
Comment (Will-Hoey) [reply core/utils.py:19]: @Iamdomchap90 I suppose my question is why do we need the form? Why can't we simply use a `create()` call?

**Pull request merged**